### ISSUE TITLE
Postsynaptic voltage change before current injection

### ISSUE THREAD

**Issue by VENOM314** [jaxleyverse/jaxley#697]
I am seeing a voltage change in postsynaptic cell before current injection; here is a minimal code that demonstrates this issue. I made sure "eLeak" is the same as "v", and I don't know what's wrong.

```
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import jaxley as jx
import jax.numpy as jnp # numpy in jax
from jaxley.channels import Leak # Mechanism
from jax import config, jit # Device
config.update("jax_platform_name", "gpu")
config.update("jax_enable_x64", True)
import os
from jaxley.synapses import IonotropicSynapse
from jaxley.connect import connect


depth = 12 # the large cell we care about is post synaptic
parents = [-1] + [b // 2 for b in range(0, 2**depth - 2)]
comp = jx.Compartment()
branch = jx.Branch(comp, 1)
cell = jx.Cell(branch, parents=parents)
cell.insert(Leak())
cell.set("Leak_gLeak", 8.0e-06)
cell.set("Leak_eLeak", -60.00) # I made sure eLeak is same as V which is correct as presynaptic cell's voltage is stable (see graph at the end)
cell.set("v", -60.00)
cell.set("axial_resistivity", 50)
cell.set("capacitance", 0.5)


synapse_locations = [(1500, 0.5)] # (which branch, which loc)


depth = 2
parents = [-1] + [b // 2 for b in range(0, 2**depth - 2)]
comp = jx.Compartment()
branch = jx.Branch(comp, 1)
stim_cell = jx.Cell(branch, parents=parents)
stim_cell.insert(Leak())
stim_cell.set("Leak_gLeak", 8.0e-06)
stim_cell.set("Leak_eLeak", -60.00)
stim_cell.set("v", -60.00)
stim_cell.set("axial_resistivity", 50)
stim_cell.set("capacitance", 0.5)


cell_list = [stim_cell, cell]
net = jx.Network(cell_list)


pre = net.cell(0).branch(0).loc(0.5)
post = net.cell(1).branch(synapse_locations[0][0]).loc(synapse_locations[0][1])
connect(pre, post, IonotropicSynapse())
net.select(edges=[0]).set("IonotropicSynapse_gS", 10) # making it extreme like 10000 gives the same result


net.delete_recordings()
net.delete_stimuli()
dt = 0.125
t_max = 250.0


currents = jx.datapoint_to_step_currents(
    i_delay = 200.0,
    i_dur= 2*dt,
    i_amp= [11.0],
    delta_t= dt,
    t_max= t_max
)

n_syn = len(synapse_locations)
stim_view = net.cell(range(n_syn)).branch(0).loc(0.5)
net.cell(n_syn).branch(0).loc(0.5).record()
net.cell(0).branch(0).loc(0.5).record()


syn_curr = currents[0]
plt.plot(syn_curr)

syn_curr = currents[0]
data_stimuli = stim_view.data_stimulate(syn_curr)
voltage = jx.integrate(net, data_stimuli=data_stimuli, delta_t=dt, t_max = t_max)


time_vector = np.arange(len(voltage[0])) * dt
fig, axs = plt.subplots(1, 2,figsize=(8, 6))
axs[0].plot(time_vector, voltage[0])
axs[1].plot(time_vector, voltage[1])
```

<img width="679" height="511" alt="Image" src="https://github.com/user-attachments/assets/8c37fa1e-106e-48ef-b190-5be6833b9086" />

The graph's x-axis is in ms while y-axis is in mv. This is a rather crude plot but as you can see current-induced change should happen after 200ms; however even before then there is a even more drastic change on postsynaptic membrane.

**Comment by jnsbck** [comment]
Thanks for reporting. I'll take a look at it. Hopefully I'll get to it in the next couple of days! If not, feel free to ping me again.

**Comment by michaeldeistler** [comment]
Alright, there are a few reasons for this:
- firstly, your synapse is _extremely_ strong: `10uS`. Because of this, even when the synapse is just a little bit "active" (i.e. the synaptic state `"IonotropicSynapse_s"` is >0, but still very small), it will lead to a large current.
- second, at `-60mV`, the synapse is actually quite a bit active: The synaptic state is around `0.1`.

Taken together, these two things leads to a very large post-synaptic depolarization.

To fix this, you have two options: First, you can simply make the synapse much weaker, for example:
```python
net.select(edges=[0]).set("IonotropicSynapse_gS", 1e-4)  # uS
```

Second, you could change the threshold at which the synapse becomes active:
```python
net.select(edges=[0]).set("IonotropicSynapse_v_th", 10.0)  # mV. Default is -35mV.
```

You can always record the synaptic state (which indicates how active the synapse is) with
```python
net.select(edges=[0]).record("IonotropicSynapse_s")
```

Finally, the synapse is initialized at `"IonotropicSynapse_s" = 0.2`. You might want to initialize it at 0.0 instead:
```python
net.set("IonotropicSynapse_s", 0.0)
```

Hope this helps!
Michael

**Comment by michaeldeistler** [comment]
This does fix it, but I don't think one wants to have a reversal around `-60.0`, because this would mean that the synapse does not add current when the presynaptic neuron is active :)
